geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/clustering/mapreduce/SimpleFeatureOutputReducer.java

```diff
@@ -0,0 +1,163 @@
+ *                name of the output SimpleFeature data type
+ * 
+ *                "SimpleFeatureOutputReducer.Global.BatchId" ->the id of the
+ *                batch; defaults to current time in millis (for range
+ *                comparisons)
+ * 
+ * 
+ * @formatter:on
+ */
+
+public class SimpleFeatureOutputReducer extends
+		GeoWaveReducer
+{
+	protected DimensionExtractor<Object> dimExtractor;
+	protected String outputDataTypeID;
+	protected String batchID;
+	protected String groupID;
+	protected FeatureDataAdapter outputAdapter;
+
+	@Override
+	protected void reduceNativeValues(
+			final GeoWaveInputKey key,
+			final Iterable<Object> values,
+			final ReduceContext<GeoWaveInputKey, ObjectWritable, GeoWaveInputKey, Object> context )
+			throws IOException,
+			InterruptedException {
+		final Iterator<Object> valIt = values.iterator();
+		if (valIt.hasNext()) {
+			key.setAdapterId(outputAdapter.getAdapterId());
+			final SimpleFeature feature = getSimpleFeature(
+					key,
+					valIt.next());
+			context.write(
+					key,
+					feature);
+		}
+	}
+
+	private SimpleFeature getSimpleFeature(
+			final GeoWaveInputKey key,
+			final Object entry ) {
+		final Geometry geometry = dimExtractor.getGeometry(entry);
+		final double[] extraDims = dimExtractor.getDimensions(entry);
+
+		final String inputID = StringUtils.stringFromBinary(key.getDataId().getBytes());
+		final SimpleFeature pointFeature = AnalyticFeature.createGeometryFeature(
+				outputAdapter.getType(),
+				batchID,
+				inputID,
+				inputID,
+				groupID,
+				0.0,
+				geometry,
+				dimExtractor.getDimensionNames(),
+				extraDims,
+				1,
+				1,
+				0);
+
+		return pointFeature;
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	protected void setup(
+			final Reducer<GeoWaveInputKey, ObjectWritable, GeoWaveInputKey, ObjectWritable>.Context context )
+			throws IOException,
+			InterruptedException {
+		super.setup(context);
+
+		outputDataTypeID = GeoWaveConfiguratorBase.getConfiguration(
+				context).get(
+				GeoWaveConfiguratorBase.enumToConfKey(
+						SimpleFeatureOutputReducer.class,
+						ExtractParameters.Extract.OUTPUT_DATA_TYPE_ID),
+				"reduced_features");
+
+		batchID = GeoWaveConfiguratorBase.getConfiguration(
+				context).get(
+				GeoWaveConfiguratorBase.enumToConfKey(
+						SimpleFeatureOutputReducer.class,
+						GlobalParameters.Global.BATCH_ID),
+				UUID.randomUUID().toString());
+
+		groupID = GeoWaveConfiguratorBase.getConfiguration(
+				context).get(
+				GeoWaveConfiguratorBase.enumToConfKey(
+						SimpleFeatureOutputReducer.class,
+						ExtractParameters.Extract.GROUP_ID),
+				UUID.randomUUID().toString());
+
+		try {
+			dimExtractor = GeoWaveConfiguratorBase.getInstance(
+					SimpleFeatureOutputReducer.class,
+					ExtractParameters.Extract.DIMENSION_EXTRACT_CLASS,
+					context,
+					DimensionExtractor.class,
+					EmptyDimensionExtractor.class);
+		}
+		catch (final Exception e1) {
+			GeoWaveReducer.LOGGER.warn(
```

Comment:
not a big deal, I just usually create a logger for the class rather than use its super class